polaris-service/src/main/java/org/apache/polaris/service/catalog/PolarisCatalogHandlerWrapper.java

```diff
@@ -301,27 +288,22 @@ private void authorizeBasicTableLikeOperationOrThrow(
 
     // The underlying Catalog is also allowed to fetch "fresh" versions of the target entity.
     resolutionManifest.addPassthroughPath(
+        PRIMARY,
         new ResolverPath(
             PolarisCatalogHelpers.tableIdentifierToList(identifier),
             PolarisEntityType.TABLE_LIKE,
             true /* optional */),
-        identifier);
+        identifier,
+        subType,
+        () -> {
+          if (subType == PolarisEntitySubType.TABLE) {
+            throw new NoSuchTableException("Table does not exist: %s", identifier);
+          } else {
+            throw new NoSuchViewException("View does not exist: %s", identifier);
+          }
```

Comment:
Nitpicking, but below we are using the ternary operator for exactly the same logic:

```suggestion
throw subType == PolarisEntitySubType.TABLE
                      ? new NoSuchTableException("Table does not exist: %s", identifier)
                      : new NoSuchViewException("View does not exist: %s", identifier);
```

Reply:
this was a pure copy-paste of old code :) I did not try to rework it for the sake of correctness (aside from the secret rotation auth, which I had to change)